source.go

```diff
@@ -574,3 +570,45 @@ func (s *Spec) getPatchedSources(sOpt SourceOpts, worker llb.State, filterFunc f
 
 	return PatchSources(worker, s, states, opts...), nil
 }
+
+// Tar creates a tar+gz from the provided state and puts it in the provided dest.
+// The provided work state is used to perform the necessary operations to produce the tarball and requires the tar and gzip binaries.
+func Tar(work, src llb.State, dest string, opts ...llb.ConstraintsOpt) llb.State {
```

Comment:
could you add the type for the `work` arg?

Reply:
Any particular reason?

Reply:
I can, actually I assumed the linter would complain with that but it doesn't seem to care.

Reply:
not strong reason just `nit` for style consistency